mteb/models/jina_models.py

```diff
@@ -213,72 +216,327 @@ def encode(
+                "Load the model with: meta.load_model(vector_type='single_vector')"
+            )
+
+        # Log task information
         prompt_name = self.get_prompt_name(self.model_prompts, task_name, prompt_type)
-        if prompt_name:
-            logger.info(
-                f"Using prompt_name={prompt_name} for task={task_name} prompt_type={prompt_type}"
+        self._log_task_info(task_name, prompt_type, prompt_name, len(sentences))
+
+        # encode() always uses single_vector mode (return_multivector=False)
+        embeddings = self.get_text_embeddings(
+            texts=list(sentences),
+            task_name=task_name,
+            batch_size=batch_size,
+            vector_type="single_vector",
+            prompt_type=prompt_type,
+            return_numpy=True,
+            **kwargs,
+        )
+
+        return embeddings
+
+    def get_text_embeddings(
+        self,
+        texts: list[str],
+        *,
+        task_name: str | None = None,
+        prompt_type: PromptType | None = None,
+        batch_size: int = 32,
+        return_numpy=False,
+        **kwargs: Any,
+    ):
+        # Resolve task parameters
+        base_task, prompt_name_param, task_type = self._resolve_task_parameters(
+            task_name, prompt_type
+        )
+
+        self._warn_vector_type_mismatch(task_type)
+
+        with torch.no_grad():
+            return self.model.encode_text(
+                texts=texts,
+                batch_size=batch_size,
+                return_multivector=self.vector_type == "multi_vector",
+                prompt_name=prompt_name_param,
+                task=base_task,
+                return_numpy=return_numpy,
             )
+
+    def get_image_embeddings(
+        self,
+        images,
+        *,
+        task_name: str | None = None,
+        prompt_type: PromptType | None = None,
+        batch_size: int = 32,
+        max_pixels: int = 37788800,
+        return_numpy=False,
+        **kwargs: Any,
+    ):
+        import torchvision.transforms.functional as F
+        from torch.utils.data import DataLoader
+
+        # Resolve task parameters
+        base_task, _, task_type = self._resolve_task_parameters(task_name, prompt_type)
+
+        self._warn_vector_type_mismatch(task_type)
+
+        all_images = []
+        if isinstance(images, DataLoader):
+            iterator = images
         else:
-            logger.info(
-                f"No model prompts found for task={task_name} prompt_type={prompt_type}"
-            )
-        logger.info(f"Encoding {len(sentences)} sentences.")
+            iterator = DataLoader(images, batch_size=batch_size)
+
+        for batch in iterator:
+            for b in batch:
+                pil_img = (
+                    F.to_pil_image(b.to("cpu")) if not isinstance(b, Image.Image) else b
+                )
+                all_images.append(pil_img)
+
+        batch_size = 1
```

Comment:
Could we not use the param that's passed in, i.e. `batch_size: int = 32`?

Reply:
Given the specified image resolution and model size, we can handle at most ~2 images per batch before running into an out-of-memory (OOM) error.